.github/workflows/test.yml

```diff
@@ -27,6 +27,12 @@ jobs:
       - uses: actions/setup-python@v5
         with:
           python-version: ${{ matrix.python-version }}
+      - uses: actions/cache@v4
+        with:
+          path: ~/.cache/pip
+          key: ${{ runner.os }}-pip-${{ hashFiles('**/requirements.txt') }}
+          restore-keys: |
+              ${{ runner.os }}-pip-
```

Comment:
Have you considered adding
`cache: pip` to the setup-python action?

see https://github.com/actions/setup-python?tab=readme-ov-file#caching-packages-dependencies

Reply:
I didn't consider that. Let me do that

Reply:
I fixed the workflow issue can you sync your fork with our main branch and do the caching change so, we merge your PR.

Reply:
also don't make the caching change until @kjaymiller comments on it maybe he prefers the other way of caching wait to see what he will say.
just update your pr with the latest changes for now.

Reply:
@john0isaac  done

Reply:
> also don't make the caching change until @kjaymiller comments on it maybe he prefers the other way of caching wait to see what he will say. just update your pr with the latest changes for now.

I agree with you John. My initial though was using the `cache: 'pip'` setting in setup-python. There's no need for us to manage another tool.